core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/FileAsyncRequestBody.java

```diff
@@ -271,34 +274,22 @@ private void closeFile() {
             }
         }
 
-        private void signalOnNext(ByteBuffer bb) {
-            synchronized (this) {
-                if (!done) {
-                    subscriber.onNext(bb);
-                }
-            }
-        }
-
         private void signalOnComplete() {
-            synchronized (this) {
-                if (!done) {
-                    subscriber.onComplete();
-                    done = true;
-                }
+            if (!done) {
+                done = true;
+                subscriber.onComplete();
             }
         }
 
         private void signalOnError(Throwable t) {
-            synchronized (this) {
```

Comment:
`synchronized` is removed because it doesn't seem to be necessary here imo.

Reply:
I believe this was needed, in case a failure happened in `request(long)` and the `inputChannel.read` simultaneously. It guards against double-sending `onError`.